model/base/src/main/java/org/eclipse/ditto/model/base/headers/DittoHeaderDefinition.java

```diff
@@ -88,10 +90,16 @@
      */
     CHANNEL("channel", String.class);
 
+    /**
+     * Map to speed up lookup of header definition by key.
+     */
+    private static final Map<CharSequence, DittoHeaderDefinition> VALUES_BY_KEY = Arrays.stream(values())
+            .collect(Collectors.toMap(DittoHeaderDefinition::getKey, Function.identity()));
+
     private final String key;
     private final Class<?> type;
 
-    private DittoHeaderDefinition(final String theKey, final Class<?> theType) {
+    DittoHeaderDefinition(final String theKey, final Class<?> theType) {
```

Comment:
IMHO there is no need to give a constructor of an enum more visibility than `private`. I know that IntelliJ has another opinion.

Reply:
enum constructors are always private, the modifier has no effect.